src/Mod/SoldierNamePool.cpp

```diff
@@ -129,6 +143,36 @@ std::string SoldierNamePool::genName(SoldierGender *gender, int femaleFrequency)
 	return name.str();
 }
 
+/**
+ * Returns a new random callsign from the
+ * lists of names contained within.
+ * @param gender Returned gender of the callsign.
+ * @return The soldier's callsign.
+ */
+std::string SoldierNamePool::genCallsign(const SoldierGender gender) const
+{
+	std::ostringstream callsign;
+	if (!_femaleCallsign.empty())
+	{
+		if (gender == GENDER_MALE)
+		{
+			size_t first = RNG::generate(0, _maleCallsign.size() - 1);
+			callsign << _maleCallsign[first];
+		}
+		else
+		{
+			size_t first = RNG::generate(0, _femaleCallsign.size() - 1);
+			callsign << _femaleCallsign[first];
+		}
+	}
+	else
+	{
+		// meaning no callsigns defined in yaml, they are optional after all
+		callsign << "*UNDEFINED*";
```

Comment:
I would prefer empty string instead of "*UNDEFINED*"

Reply:
Will do